wolfSSL/src/chapter02.md

```diff
@@ -680,15 +1234,325 @@ Enable ECC SECP384R1 Single-Precision (SP) support. Only applies to `WOLFSSL_SP_
+
+#### WOLFSSL_DTLS13
+
+Enable wolfSSL DTLS 1.3.
+
+#### WOLFSSL_TLS13
+
+Enable TLS 1.3 protocol implementation.
+
+#### WOLFSSL_TLS13_IGNORE_AEAD_LIMITS
+
+Limits specified by https://www.rfc-editor.org/rfc/rfc9147.html#name-aead-limits. We specify the limit by which we need to do a key update as the halfway point to the hard decryption fail limit.
+
+#### WOLFSSL_TLS13_MIDDLEBOX_COMPAT
+
+Enable middlebox compatibility in the TLS 1.3 handshake. This includes sending ChangeCipherSpec before encrypted messages and including a session ID.
+
+#### WOLFSSL_TLS13_SHA512
+
+Allow generation of SHA-512 digests in a handshake - no ciphersuite. Requires SHA-512 at this time.
+
+#### WOLFSSL_UIP
+
+When CONTIKI is defined, it is an implementation of UIP.
+
+#### TLS13_MAX_TICKET_AGE
+
+Specifies Max ticket age. For TLS 1.3, this is 7 days.
+
+#### TLS13_TICKET_NONCE_STATIC_SZ
+
+TLS13_TICKET_NONCE_STATIC_SZ is not supported in this FIPS_VERSION_GE.
+
+#### TLS13_TICKET_NONCE_MAX_SZ
+
+Defines version max size for ticket nonce. Max size is defined as 255 bytes.
+
+#### WOLFSSL_TICKET_ENC_AES128_GCM
+
+Use AES128-GCM to encrypt/decrypt session tickets in the default callback. This is server-only. If ChaCha20/Poly1305 is not compiled, this is the default algorithm.
+
+#### WOLFSSL_TICKET_ENC_AES256_GCM
+
+Use AES256-GCM to encrypt/decrypt session tickets in default callback. Server only.
+
+#### WOLFSSL_TICKET_ENC_CHACHA20_POLY1305
+
+Use ChaCha20-Poly1305 to encrypt/decrypt session tickets in the default callback. If none are defined, the default algorithm is used, and algorithms are compiled. This is server-only.
+
+#### WOLFSSL_TICKET_EXTRA_PADDING_SZ
+
+Defines ticket extra padding size defined as 32.
+
+#### WOLFSSL_TICKET_HAVE_ID
+
+Use to make sure the ticket has ID. Only add to the cache when support is built in and when the ticket contains an ID. Otherwise we have no way to actually retrieve a ticket from the cache.
+
+#### WOLFSSL_TICKET_KEY_LIFETIME
+
+The default lifetime is 1 hour from the issue of the first ticket with the key. It must be longer than a hint.
+
+#### WOLFSSL_TICKET_NONCE_MALLOC
+
+Enable dynamic allocation of ticket nonces. Need to disable the HKDF expand callbacks.
+
+#### SHOW_CERTS
+
+Show certs will output certs when defined. Use for embedded debugging.
+
+#### SHOW_SECRETS
+
+Used for debugging. It will show applicable secrets.
+
+#### DEBUG_UNIT_TEST_CERTS
+
+Used when debugging name constraint tests. Not static to allow use in multiple locations with complex define guards.
+
+#### DEBUG_WOLFSSL_VERBOSE
+
+When using the OPENSSL_EXTRA or DEBUG_WOLFSSL_VERBOSE macro, WOLFSSL_ERROR is mapped to the new function WOLFSSL_ERROR_LINE, which gets the line number and function name where WOLFSSL_ERROR is called.
+
+#### SOCKET_INVALID
+
+Used to define an invalid socket and is defined as -1.
+
+#### WOLFSSL_SOCKET_INVALID
+
+Used in socket handling.
```

Comment:
For testing only allows overriding the value used to indicate an invalid socket. Typically is `-1`.